index.js

```diff
@@ -42,7 +42,7 @@ module.exports = function (req, time) {
 		if (delays.socket !== undefined) {
 			// Abort the request if there is no activity on the socket for more
 			// than `delays.socket` milliseconds.
-			this.setTimeout(delays.socket, function socketTimeoutHandler() {
+			req.setTimeout(delays.socket, function socketTimeoutHandler() {
```

Comment:
We do that, the `connect()` is a listener for the `connect` event, so when we do this, the socket is connected.